### ISSUE TITLE
Error on watch branch 

### ISSUE THREAD

**Issue by poquirion** [jts/ncov-tools#61]
We are testing the tool with the watch branch and are getting this hard to track error:
```
48 of 70 steps (69%) done
Select jobs to execute...
Failed to solve scheduling problem with ILP solver. Falling back to greedy solver.Run Snakemake with --verbose to see the full solver output for debugging the problem.
```

It points to missing outputs. We do not have that error with the latest release (1.3.1)

If that helps, here is the list of software in our conda environement. Note that depending on when you build the conda env, this sofware list could be different. I think it would make sens to have a version for all package in the `environment.yml` file to try to limit the variation.

```$ conda list
# packages in environment at /opt/conda/envs/ncov-qc:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
_r-mutex                  1.0.1               anacondar_1    conda-forge
aioeasywebdav             2.4.0           py37hc8dfbb8_1001    conda-forge
aiohttp                   3.7.3            py37h5e8e339_1    conda-forge
alsa-lib                  1.2.3                h516909a_0    conda-forge
amply                     0.1.4                      py_0    conda-forge
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
async-timeout             3.0.1                   py_1000    conda-forge
attmap                    0.12.11                    py_0    conda-forge
attrs                     20.3.0             pyhd3deb0d_0    conda-forge
augur                     11.0.0                     py_0    bioconda
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
bcbio-gff                 0.6.6              pyh864c0ab_1    bioconda
bcrypt                    3.2.0            py37h5e8e339_1    conda-forge
bedtools                  2.30.0               hc088bd4_0    bioconda
binutils_impl_linux-64    2.35.1               h193b22a_1    conda-forge
binutils_linux-64         2.35                hc3fd857_29    conda-forge
bioconductor-ggtree       2.0.0                     r36_0    bioconda
bioconductor-treeio       1.10.0                    r36_0    bioconda
biopython                 1.77             py37h8f50634_1    conda-forge
boto3                     1.16.59            pyhd8ed1ab_0    conda-forge
botocore                  1.19.59            pyhd8ed1ab_0    conda-forge
brotlipy                  0.7.0           py37h5e8e339_1001    conda-forge
bwidget                   1.9.14               ha770c72_0    conda-forge
bx-python                 0.8.9            py37h73d7ac5_2    bioconda
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.17.1               h36c2ea0_0    conda-forge
ca-certificates           2020.12.5            ha878542_0    conda-forge
cached-property           1.5.2                    pypi_0    pypi
cachetools                4.2.1              pyhd8ed1ab_0    conda-forge
cairo                     1.16.0            hcf35c78_1003    conda-forge
certifi                   2020.12.5        py37h89c1867_1    conda-forge
cffi                      1.14.4           py37h11fe52a_0    conda-forge
chardet                   3.0.4           py37he5f6b98_1008    conda-forge
click                     7.1.2                    pypi_0    pypi
coincbc                   2.10.5               hab63836_1    conda-forge
configargparse            1.2.3              pyh9f0ad1d_0    conda-forge
cryptography              3.3.1            py37h7f0c10b_1    conda-forge
curl                      7.68.0               hf8cf82a_0    conda-forge
cvxopt                    1.2.4           py37hcf54b7c_201    conda-forge
cycler                    0.10.0                     py_2    conda-forge
datafunk                  0.0.8                    pypi_0    pypi
datapackage               1.15.1                   pypi_0    pypi
datrie                    0.8.2            py37h8f50634_1    conda-forge
decorator                 4.4.2                      py_0    conda-forge
dendropy                  4.5.1                    pypi_0    pypi
docutils                  0.16             py37h89c1867_3    conda-forge
dropbox                   10.9.0             pyhd3deb0d_0    conda-forge
dsdp                      5.8               hd9d9efa_1203    conda-forge
epiweeks                  2.1.2                    pypi_0    pypi
et-xmlfile                1.0.1                    pypi_0    pypi
expat                     2.2.9                h9c3ff4c_3    conda-forge
fasttree                  2.1.10               h516909a_4    bioconda
fftw                      3.3.8           nompi_hfc0cae8_1114    conda-forge
filechunkio               1.8                        py_2    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      2.001                hab24e00_0    conda-forge
font-ttf-source-code-pro  2.030                hab24e00_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.13.1            hba837de_1004    conda-forge
fonts-conda-forge         1                             0    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
fribidi                   1.0.10               h36c2ea0_0    conda-forge
ftputil                   4.0.0                      py_0    conda-forge
gcc_impl_linux-64         7.5.0               habd7529_18    conda-forge
gcc_linux-64              7.5.0               he2a3fca_29    conda-forge
gdk-pixbuf                2.42.2               h19a9c64_1    conda-forge
gettext                   0.19.8.1          hf34092f_1004    conda-forge
gfortran_impl_linux-64    7.5.0               h56cb351_18    conda-forge
gfortran_linux-64         7.5.0               ha081f1e_29    conda-forge
ghostscript               9.53.3               h58526e2_2    conda-forge
giflib                    5.2.1                h36c2ea0_2    conda-forge
gitdb                     4.0.5                      py_0    conda-forge
gitpython                 3.1.12             pyhd8ed1ab_0    conda-forge
glib                      2.66.3               h58526e2_0    conda-forge
glpk                      4.65              h9202a9a_1003    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
google-api-core           1.24.1             pyhd3deb0d_0    conda-forge
google-api-python-client  1.12.8             pyhd3deb0d_0    conda-forge
google-auth               1.24.0             pyhd3deb0d_0    conda-forge
google-auth-httplib2      0.0.4              pyh9f0ad1d_0    conda-forge
google-cloud-core         1.5.0              pyhd3deb0d_0    conda-forge
google-cloud-storage      1.35.0             pyhd3deb0d_0    conda-forge
google-crc32c             1.1.0            py37hab72019_0    conda-forge
google-resumable-media    1.2.0              pyhd3deb0d_0    conda-forge
googleapis-common-protos  1.52.0           py37h89c1867_1    conda-forge
graphite2                 1.3.13            h58526e2_1001    conda-forge
graphviz                  2.42.3               h0511662_0    conda-forge
grpcio                    1.34.1           py37hb27c1af_0    conda-forge
gsl                       2.5                  h294904e_1    conda-forge
gxx_impl_linux-64         7.5.0               hd0bb8aa_18    conda-forge
gxx_linux-64              7.5.0               h547f3ba_29    conda-forge
harfbuzz                  2.4.0                h9f30f68_3    conda-forge
htslib                    1.9                  h4da6232_3    bioconda
httplib2                  0.18.1             pyh9f0ad1d_0    conda-forge
icu                       64.2                 he1b5a44_1    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
ijson                     3.1.3                    pypi_0    pypi
imagemagick               7.0.10_28       pl5262hd815ba0_2    conda-forge
importlib-metadata        3.4.0            py37h89c1867_0    conda-forge
importlib_metadata        3.4.0                hd8ed1ab_0    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
iqtree                    1.6.12               he513fc3_1    bioconda
isodate                   0.6.0                    pypi_0    pypi
jbig                      2.1               h516909a_2002    conda-forge
jdcal                     1.4.1                    pypi_0    pypi
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
jmespath                  0.10.0             pyh9f0ad1d_0    conda-forge
joblib                    1.0.0                    pypi_0    pypi
jpeg                      9d                   h36c2ea0_0    conda-forge
jsonlines                 2.0.0                    pypi_0    pypi
jsonpointer               2.0                      pypi_0    pypi
jsonschema                3.2.0                      py_2    conda-forge
jupyter_core              4.7.0            py37h89c1867_1    conda-forge
k8                        0.2.5                he513fc3_0    bioconda
kernel-headers_linux-64   2.6.32              h77966d4_13    conda-forge
kiwisolver                1.3.1            py37h2527ec5_1    conda-forge
krb5                      1.16.4               h2fd8d38_0    conda-forge
lcms2                     2.11                 hcbb858e_1    conda-forge
ld_impl_linux-64          2.35.1               hea4e1c9_1    conda-forge
libblas                   3.9.0                7_openblas    conda-forge
libcblas                  3.9.0                7_openblas    conda-forge
libcrc32c                 1.1.1                h9c3ff4c_2    conda-forge
libcurl                   7.68.0               hda55be3_0    conda-forge
libdeflate                1.6                  h516909a_0    conda-forge
libedit                   3.1.20191231         h46ee950_2    conda-forge
libffi                    3.2.1             he1b5a44_1007    conda-forge
libgcc-devel_linux-64     7.5.0               hda03d7c_18    conda-forge
libgcc-ng                 9.3.0               h2828fa1_18    conda-forge
libgfortran-ng            7.5.0               h14aa051_18    conda-forge
libgfortran4              7.5.0               h14aa051_18    conda-forge
libglib                   2.66.3               hbe7bbb4_0    conda-forge
libgomp                   9.3.0               h2828fa1_18    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
liblapack                 3.9.0                7_openblas    conda-forge
libopenblas               0.3.12          pthreads_hb3c22a3_1    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libprotobuf               3.14.0               h780b84a_0    conda-forge
librsvg                   2.50.2               h3442318_1    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libssh2                   1.9.0                hab1572f_5    conda-forge
libstdcxx-devel_linux-64  7.5.0               hb016644_18    conda-forge
libstdcxx-ng              9.3.0               h6de172a_18    conda-forge
libtiff                   4.2.0                hdc55705_0    conda-forge
libtool                   2.4.6             h58526e2_1007    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libwebp                   1.1.0                h76fa15c_4    conda-forge
libwebp-base              1.1.0                h36c2ea0_3    conda-forge
libxcb                    1.13              h7f98852_1003    conda-forge
libxml2                   2.9.10               hee79883_0    conda-forge
lineages                  2020-05-19-2             pypi_0    pypi
linear-tsv                1.1.0                    pypi_0    pypi
logmuse                   0.2.6              pyh8c360ce_0    conda-forge
lz4-c                     1.9.3                h9c3ff4c_0    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
mafft                     7.475                h516909a_0    bioconda
make                      4.3                  hd18ef5c_1    conda-forge
markupsafe                1.1.1            py37h5e8e339_3    conda-forge
matplotlib-base           3.3.3            py37h0c9df89_0    conda-forge
metis                     5.1.0             h58526e2_1006    conda-forge
minimap2                  2.17                 hed695b0_3    bioconda
more-itertools            8.6.0              pyhd8ed1ab_0    conda-forge
multidict                 5.1.0            py37h5e8e339_1    conda-forge
nbformat                  5.1.2              pyhd8ed1ab_1    conda-forge
ncov-parser               0.6.3                    pypi_0    pypi
ncov-watch                0.0.1                    pypi_0    pypi
ncurses                   6.1               hf484d3e_1002    conda-forge
networkx                  2.5                        py_0    conda-forge
newick_utils              1.6                  h516909a_3    bioconda
numpy                     1.19.1           py37h7ea13bd_2    conda-forge
oauth2client              4.1.3                      py_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjdk                   11.0.8               hacce0ff_0    conda-forge
openjpeg                  2.3.1                hf7af979_3    conda-forge
openpyxl                  3.0.6                    pypi_0    pypi
openssl                   1.1.1i               h7f98852_0    conda-forge
packaging                 20.8               pyhd3deb0d_0    conda-forge
pandas                    1.2.1            py37hdc94413_0    conda-forge
pango                     1.42.4               h7062337_4    conda-forge
pangolearn                2021-01-22               pypi_0    pypi
pangolin                  2.1.8                    pypi_0    pypi
paramiko                  2.7.2              pyh9f0ad1d_0    conda-forge
parasail-python           1.2.3            py37h13b99d1_0    bioconda
pcre                      8.44                 he1b5a44_0    conda-forge
peppy                     0.31.0             pyh9f0ad1d_0    conda-forge
perl                      5.26.2            h36c2ea0_1008    conda-forge
pillow                    8.1.0            py37he6b4880_1    conda-forge
pip                       19.3.1                   py37_0    conda-forge
pixman                    0.38.0            h516909a_1003    conda-forge
pkg-config                0.29.2            h36c2ea0_1008    conda-forge
pluggy                    0.13.1           py37h89c1867_4    conda-forge
poppler                   0.87.0               h9584818_0    conda-forge
poppler-data              0.4.10                        0    conda-forge
prettytable               2.0.0              pyhd8ed1ab_0    conda-forge
protobuf                  3.14.0           py37hcd2ae1e_1    conda-forge
psutil                    5.8.0            py37h5e8e339_1    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
pulp                      2.3.1            py37hc8dfbb8_0    conda-forge
py                        1.10.0             pyhd3deb0d_0    conda-forge
pyasn1                    0.4.8                      py_0    conda-forge
pyasn1-modules            0.2.7                      py_0    conda-forge
pycountry                 20.7.3                   pypi_0    pypi
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pygments                  2.7.4              pyhd8ed1ab_0    conda-forge
pygraphviz                1.6              py37h0b1d2a2_1    conda-forge
pynacl                    1.4.0            py37h5e8e339_2    conda-forge
pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyrsistent                0.17.3           py37h5e8e339_2    conda-forge
pysam                     0.16.0           py37ha9a96c6_0    bioconda
pysftp                    0.2.9                      py_1    conda-forge
pysocks                   1.7.1            py37h89c1867_3    conda-forge
pytest                    6.2.1            py37h89c1867_1    conda-forge
python                    3.7.8           h6f2ec95_1_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python-irodsclient        0.8.2                      py_0    conda-forge
python-lzo                1.12            py37he0a3664_1003    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
pytools                   2020.1                   pypi_0    pypi
pytz                      2020.5             pyhd8ed1ab_0    conda-forge
pyvcf                     0.6.8                    pypi_0    pypi
pyyaml                    5.4.1            py37h5e8e339_0    conda-forge
r-ape                     5.4_1             r36h51c796c_0    conda-forge
r-argparse                2.0.3             r36hc72bb7e_1    conda-forge
r-assertthat              0.2.1             r36h6115d3f_2    conda-forge
r-backports               1.2.1             r36hcfec24a_0    conda-forge
r-base                    3.6.1                h3a67422_6    conda-forge
r-biocmanager             1.30.10           r36h6115d3f_1    conda-forge
r-brio                    1.1.1             r36hcfec24a_0    conda-forge
r-callr                   3.5.1             r36h142f84f_0    conda-forge
r-cli                     2.2.0             r36hc72bb7e_0    conda-forge
r-colorspace              2.0_0             r36h9e2df91_0    conda-forge
r-crayon                  1.3.4           r36h6115d3f_1003    conda-forge
r-desc                    1.2.0           r36h6115d3f_1003    conda-forge
r-diffobj                 0.3.3             r36hcfec24a_0    conda-forge
r-digest                  0.6.27            r36h1b71b39_0    conda-forge
r-dplyr                   1.0.3             r36h03ef668_0    conda-forge
r-ellipsis                0.3.1             r36hcdcec82_0    conda-forge
r-evaluate                0.14              r36h6115d3f_2    conda-forge
r-fansi                   0.4.2             r36hcfec24a_0    conda-forge
r-farver                  2.0.3             r36h0357c0b_1    conda-forge
r-findpython              1.0.5             r36h6115d3f_2    conda-forge
r-generics                0.1.0             r36hc72bb7e_0    conda-forge
r-getopt                  1.20.3                    r36_2    conda-forge
r-ggforce                 0.3.1             r36h0357c0b_1    conda-forge
r-ggplot2                 3.3.3             r36hc72bb7e_0    conda-forge
r-glue                    1.4.2             r36hcfec24a_0    conda-forge
r-gtable                  0.3.0             r36h6115d3f_3    conda-forge
r-isoband                 0.2.3             r36h03ef668_0    conda-forge
r-jsonlite                1.7.2             r36hcfec24a_0    conda-forge
r-labeling                0.4.2             r36h142f84f_0    conda-forge
r-lattice                 0.20_41           r36hcfec24a_3    conda-forge
r-lazyeval                0.2.2             r36hcdcec82_2    conda-forge
r-lifecycle               0.2.0             r36h6115d3f_1    conda-forge
r-magrittr                2.0.1             r36hcfec24a_1    conda-forge
r-mass                    7.3_53            r36hcfec24a_0    conda-forge
r-matrix                  1.3_2             r36he454529_0    conda-forge
r-mgcv                    1.8_33            r36h7fa42b6_0    conda-forge
r-munsell                 0.5.0           r36h6115d3f_1003    conda-forge
r-nlme                    3.1_150           r36h31ca83e_0    conda-forge
r-patchwork               1.0.0             r36h6115d3f_1    conda-forge
r-pillar                  1.4.7             r36hc72bb7e_0    conda-forge
r-pkgbuild                1.2.0             r36hc72bb7e_0    conda-forge
r-pkgconfig               2.0.3             r36h6115d3f_1    conda-forge
r-pkgload                 1.1.0             r36h0357c0b_0    conda-forge
r-polyclip                1.10_0            r36h0357c0b_2    conda-forge
r-praise                  1.0.0           r36h6115d3f_1004    conda-forge
r-prettyunits             1.1.1             r36h6115d3f_1    conda-forge
r-processx                3.4.5             r36hcfec24a_0    conda-forge
r-ps                      1.5.0             r36hcfec24a_0    conda-forge
r-purrr                   0.3.4             r36hcdcec82_1    conda-forge
r-r6                      2.5.0             r36hc72bb7e_0    conda-forge
r-rcolorbrewer            1.1_2           r36h6115d3f_1003    conda-forge
r-rcpp                    1.0.6             r36h03ef668_0    conda-forge
r-rcppeigen               0.3.3.9.1         r36h306847c_0    conda-forge
r-rematch2                2.1.2             r36h6115d3f_1    conda-forge
r-rlang                   0.4.10            r36hcfec24a_0    conda-forge
r-rprojroot               2.0.2             r36hc72bb7e_0    conda-forge
r-rstudioapi              0.13              r36hc72bb7e_0    conda-forge
r-rvcheck                 0.1.8             r36h6115d3f_1    conda-forge
r-scales                  1.1.1             r36h6115d3f_0    conda-forge
r-testthat                3.0.1             r36h03ef668_0    conda-forge
r-tibble                  3.0.5             r36hcfec24a_0    conda-forge
r-tidyr                   1.1.2             r36h0357c0b_0    conda-forge
r-tidyselect              1.1.0             r36h6115d3f_0    conda-forge
r-tidytree                0.3.3             r36h6115d3f_1    conda-forge
r-tweenr                  1.0.1           r36h0357c0b_1002    conda-forge
r-utf8                    1.1.4           r36hcdcec82_1003    conda-forge
r-vctrs                   0.3.6             r36hcfec24a_0    conda-forge
r-viridislite             0.3.0           r36h6115d3f_1003    conda-forge
r-waldo                   0.2.3             r36hc72bb7e_0    conda-forge
r-withr                   2.4.0             r36hc72bb7e_0    conda-forge
r-zeallot                 0.1.0           r36h6115d3f_1002    conda-forge
ratelimiter               1.2.0                   py_1002    conda-forge
raxml                     8.2.12               h516909a_2    bioconda
readline                  8.0                  h46ee950_1    conda-forge
requests                  2.25.1             pyhd3deb0d_0    conda-forge
rfc3986                   1.4.0                    pypi_0    pypi
rsa                       4.7                pyhd3deb0d_0    conda-forge
s3transfer                0.3.4              pyhd8ed1ab_0    conda-forge
samtools                  1.9                 h10a08f8_12    bioconda
scikit-learn              0.23.1                   pypi_0    pypi
scipy                     1.5.3            py37h8911b10_0    conda-forge
sed                       4.8                  he412f7d_0    conda-forge
setuptools                49.6.0           py37h89c1867_3    conda-forge
simplejson                3.17.2           py37h5e8e339_2    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
slacker                   0.14.0                     py_0    conda-forge
smmap                     4.0.0              pyh44b312d_0    conda-forge
snakemake                 5.32.0                        0    bioconda
snakemake-minimal         5.32.0                     py_0    bioconda
snpeff                    5.0                           0    bioconda
sqlalchemy                1.3.22                   pypi_0    pypi
sqlite                    3.32.3               hcee41ef_1    conda-forge
suitesparse               5.7.2                h717dc36_0    conda-forge
sysroot_linux-64          2.12                h77966d4_13    conda-forge
tableschema               1.20.0                   pypi_0    pypi
tabulator                 1.53.1                   pypi_0    pypi
tbb                       2020.2               h4bd325d_3    conda-forge
threadpoolctl             2.1.0                    pypi_0    pypi
tk                        8.6.10               h21135ba_1    conda-forge
tktable                   2.10                 hb7b940f_3    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
toposort                  1.6                pyhd8ed1ab_0    conda-forge
tornado                   6.1              py37h5e8e339_1    conda-forge
traitlets                 5.0.5                      py_0    conda-forge
treetime                  0.8.1              pyh3252c3a_0    bioconda
typing-extensions         3.7.4.3                       0    conda-forge
typing_extensions         3.7.4.3                    py_0    conda-forge
ubiquerg                  0.6.1              pyh9f0ad1d_0    conda-forge
unicodecsv                0.14.1                   pypi_0    pypi
unidecode                 1.1.2                    pypi_0    pypi
uritemplate               3.0.1                      py_0    conda-forge
urllib3                   1.26.2             pyhd8ed1ab_0    conda-forge
vcftools                  0.1.16               he513fc3_4    bioconda
veracitools               0.1.3                      py_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
wrapt                     1.12.1           py37h5e8e339_3    conda-forge
xlrd                      2.0.1                    pypi_0    pypi
xmlrunner                 1.7.7                      py_0    conda-forge
xorg-fixesproto           5.0               h14c3975_1002    conda-forge
xorg-inputproto           2.3.2             h7f98852_1002    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.12               h516909a_0    conda-forge
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxfixes            5.0.3             h516909a_1004    conda-forge
xorg-libxi                1.7.10               h516909a_0    conda-forge
xorg-libxpm               3.5.13               h516909a_0    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-libxt                1.1.5             h516909a_1003    conda-forge
xorg-libxtst              1.2.3             h516909a_1002    conda-forge
xorg-recordproto          1.14.2            h516909a_1002    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h7f98852_1002    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
yaml                      0.2.5                h516909a_0    conda-forge
yarl                      1.5.1            py37h8f50634_0    conda-forge
zipp                      3.4.0                      py_0    conda-forge
zlib                      1.2.11            h516909a_1010    conda-forge
zstd                      1.4.8                ha95c52a_1    conda-forge
```

**Comment by jts** [comment]
This is possibly a regression in Snakemake: https://github.com/snakemake/snakemake/issues/771

I am using 5.23.0 and don't have this problem. I am tied up at the moment but will update and try to reproduce this with 5.32.0.

**Comment by poquirion** [comment]
we could also set the gihub pip install to a specific commit:
```$ pip install git+git://github.com/<user>/<repo>.git@<hash>```

**Comment by poquirion** [comment]
We will start with building with the latest snakemake version that was working on our side (5.31.1), then @pphector, maybe you can give log info.

**Comment by fmaguire** [comment]
It's weird that the scheduler is failing with such a simple workflow that doesn't have any steps with undefined output (e.g., `checkpoint` stages).

**Comment by pphector** [comment]
Ok, the actual failing job is here: 

```
MissingInputException in line 71 of /app/rules/annotation.smk:
Missing input files for rule run_snpeff:
data/L00318826.pass.vcf.gz
```

It seems like it is not finding the VCF file to run with SnpEff, however the VCF file is there, it is just not compressed. Is the `pass.vcf.gz` part hardcoded in the SnpEff call?

**Comment by pphector** [comment]
Here is a copy of the YAML file, in it the pattern for the VCFs is properly defined, and it does not have the `pass.vcf.gz` ending that the error refers to. It is the same YAML file we have used for ONT runs of the previous version of ncov tools. 

```
# path to the top-level directory containing the analysis results
data_root: data

# optionally the plots can have a "run name" prefix. If this is not defined the prefix will be "default"
run_name: LSPQoutbreak917

# path to the file containing the amplicon regions (not the primer sites, the actual amplicons)
amplicon_bed: /genfs/projects/analyste_dev/artic_network/ncov-tools/etc/nCov-2019.V3.regions.bed

# path to the nCov reference genome
reference_genome: /cvmfs/ref.mugqic/genomes/species/Coronavirinae.SARS-CoV-2/genome/Coronavirinae.SARS-CoV-2.fa

# the sequencing platform used, can be "oxford-nanopore" or "illumina"
platform: "oxford-nanopore"

# path to the BED file containing the primers, this should follow the format downloaded from
# the ARTIC primer
primer_bed: /genfs/projects/analyste_dev/artic_network/artic-ncov2019/primer_schemes/nCoV-2019/V3/nCoV-2019.bed

# offset for the amplicons and primers
offset: 0

# minimum completeness threshold for inclusion to the SNP tree plot, if no entry
# is provided the default is set to 0.75
completeness_threshold: 0.9

# the naming convention for the bam files
# this can use the variables {data_root} (as above) and {sample}
# As per the example above, this will expand to run_200431/sampleA.sorted.bam for sampleA
bam_pattern: "{data_root}/{sample}.mapped.primertrimmed.sorted.bam"

# the naming convention for the consensus sequences
consensus_pattern: "{data_root}/{sample}.consensus.fasta"

# the naming convention for the variants file, NF illumina runs typically use
# "{data_root}/{sample}.variants.vcf and oxford nanopore runs use "{data_root}/{sample}.pass.vcf.gz"
variants_pattern: "{data_root}/{sample}.variants.vcf"

metadata: "metadata.tsv"

#
# if a list of sample IDs for negative controls is provided, a report containing the amount
# of coverage detected in the negative controls can be generated
negative_control_samples: [ "NegCtrl_Ext_LSPQ917", "NegCtrl_RT_LSPQ917" ]

#
# when building a tree of the consensus genomes you can optionally include other sequences
# in the tree by providing a fasta file here
# tree_include_consensus: some_genomes_from_gisaid.fasta

#
# set this flag to true to include lineage assignments with pangolin in the output plots
assign_lineages: true
```

**Comment by jts** [comment]
Ah, @rdeborja I think this needs to prefer the variants pattern defined in the config rather than the platform-assumed pattern:

https://github.com/jts/ncov-tools/blob/511f5efbdb9f3f13b37529cfe287408f30b0a0a7/workflow/rules/annotation.smk#L12

**Comment by rdeborja** [comment]
@poquirion I fixed the hardcoded file patterns and instead uses the `variants_pattern:` parameter in the config file.  Can you try branch `fix_snpeff_dependency`.

**Comment by jts** [comment]
Closed by #67
